twython/endpoints.py

```diff
@@ -144,6 +144,10 @@ def upload_media(self, **params):
         """
         return self.post('https://upload.twitter.com/1.1/media/upload.json', params=params)
 
+    def upload_media_with_file(self, filename):
+        with open(filename, 'rb') as f:
+            return self.upload_media(media=f)
```

Comment:
Can we do something like this?

```python
content = None
_file = open(filename, 'rb')
with _file:
    content = self.upload_media(media=_file)

_file.close()
return content
```

Other versions of Python (not CPython) don't automatically close references.

See http://stackoverflow.com/questions/19039704/is-python-retaining-a-reference-to-a-file-opened-in-a-list